### ISSUE TITLE
clip high doesn't work according to your code

### ISSUE THREAD

**Issue by Life-0-1** [agentica-project/rllm#81]
https://github.com/agentica-project/rllm/blob/main/verl/verl/trainer/ppo/core_algos.py#L217

the correct code should be: `pg_loss = verl_F.masked_mean(pg_loss, eos_mask)`?

if you use `pg_loss = verl_F.masked_mean(pg_losses, eos_mask)`,  ` pg_losses2` is never considered.

so, how did you conclude that clip high is better in your blog?

`    

    pg_losses = -advantages * ratio
    
    low_clip = cliprange_low if cliprange_low != -1 else cliprange
    high_clip = cliprange_high if cliprange_high != -1 else cliprange

    pg_losses2 = -advantages * torch.clamp(ratio, 1.0 - low_clip, 1.0 + high_clip)

    pg_loss = torch.max(pg_losses, pg_losses2)
    
    if use_token_level_loss:
        pg_loss = verl_F.masked_mean(pg_losses, eos_mask)
    else:
        pg_loss = torch.sum(pg_losses * eos_mask, dim=1) / seq_len_per_sample
        pg_loss = torch.mean(pg_loss)    

    pg_clipfrac = verl_F.masked_mean(torch.gt(pg_losses2, pg_losses).float(), eos_mask)
`

**Comment by michaelzhiluo** [comment]
Good catch; it's now fixed ;) It was a bug with porting stuff over
